In [5]:
import pandas as pd

Open the excel file, pandas loads the first sheet by default.

In [2]:
df = pd.read_excel('World_Development_Indicators.xlsx')

In [3]:
df.count()

Series Name      266
Series Code      264
Country Name     264
Country Code     264
1990 [YR1990]    264
2000 [YR2000]    264
2009 [YR2009]    264
2010 [YR2010]    264
2011 [YR2011]    264
2012 [YR2012]    264
2013 [YR2013]    264
2014 [YR2014]    264
2015 [YR2015]    264
2016 [YR2016]    264
2017 [YR2017]    264
2018 [YR2018]    264
dtype: int64

Why the length of 'Series Name' is 266 ? Let us see the value_counts of 'Series Name'

In [6]:
df['Series Name'].value_counts()

GDP (current US$)                                   264
Last Updated: 03/21/2019                              1
Data from database: World Development Indicators      1
Name: Series Name, dtype: int64

The read_excel method will stop when there is not text.You can open the excel to check the result.